modules/vye/app/models/concerns/vye/gen_digest.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'openssl'
+
+module Vye
+  module GenDigest
+    extend ActiveSupport::Concern
+
+    module Common
+      GEN_DIGEST_CONFIG =
+        [Settings.vye.scrypt, %i[salt N r p length]]
+        .then { |s, l| l.collect { |m| [m, s.send(m)] } }
+        .then { |pairs| Hash[*pairs.flatten] }
+        .freeze
```

Comment:
You can get the same thing by doing 
```ruby
GEN_DIGEST_CONFIG = Settings.vye.scrypt.to_h.freeze
```

However, I do question why `N`, `r`, `p`, and `length` need to be settings. Are they going to be different across environments?

Reply:
These values do effect memory and CPU load, and can be fine tuned for the environment.